### PR TITLE
Implement ClientClusterService.getClusterVersion() [HZ-4446]

### PR DESCRIPTION
**Pull request by burakgok**
Mono PR: hazelcast/hazelcast-mono#871

### CODE REVIEW

**Review by srknzl — COMMENTED** [review]
cpp, cs and ts language generations fail. can you fix them?

**Comment by burakgok** [comment]
verify

**Comment by burakgok** [comment]
verify

**Comment by burakgok** [comment]
verify

File: protocol-definitions/custom/Custom.yaml [hunk 1]
```diff
@@ -1499,3 +1499,15 @@ customTypes:
         type: String
         since: 2.7
         nullable: true
+  - name: Version
+    since: 2.7
+    returnWithFactory: true
+    params:
```
Comment (ihsandemir) [inline protocol-definitions/custom/Custom.yaml:1505]: Very similar to `MemberVersion` (assume patch field is 0). Do we really need yet another new custom type?

File: protocol-definitions/custom/Custom.yaml [hunk 2]
```diff
@@ -1499,3 +1499,15 @@ customTypes:
         type: String
         since: 2.7
         nullable: true
+  - name: Version
```
Comment (ihsandemir) [inline protocol-definitions/custom/Custom.yaml:1502]:
```suggestion
  - name: ClusterVersion
```

File: protocol-definitions/custom/Custom.yaml (hunk 1 above)
Comment (burakgok) [reply protocol-definitions/custom/Custom.yaml:1505]: The cluster version is not `MemberVersion`, and I am not introducing another type, just adding an existing type to the client protocol. The WAN version is also a `Version`, i.e. it consists of major and minor versions only, so `Version` may have other usages in the _near_ future, considering that we try to send more cluster information to the clients.

File: protocol-definitions/custom/Custom.yaml (hunk 2 above)
Comment (burakgok) [reply protocol-definitions/custom/Custom.yaml:1502]: https://github.com/hazelcast/hazelcast-client-protocol/pull/511#discussion_r1519414404

**Review by srknzl — APPROVED** [review]

**Review by JamesHazelcast — APPROVED** [review]

File: protocol-definitions/custom/Custom.yaml [hunk 3]
```diff
@@ -1499,3 +1499,15 @@ customTypes:
...
+    since: 2.7
```
Comment (ihsandemir) [inline protocol-definitions/custom/Custom.yaml:1503]:
```suggestion
    since: 2.8
```
For 5.5 release, we need to increment the version.

Comment (burakgok) [reply protocol-definitions/custom/Custom.yaml]: 74593a07

**Pull request merged**